### ISSUE TITLE
Testkits with unidriver

### ISSUE THREAD

**Issue by mosheKerbel** [wix/wix-style-react#4333]
# ✨ Feature Request

This is more of a discussion

### 🔦 Scope

Testkits

### 💁 Explanation

We added to ~ 90% of the old wsr components uni.driver files
New components have only uni.driver files
ReactBase was refactored to a minimal usage, but got some small work left there (out of scope).

New components like `Stepper` export all testkit platforms using `stepper.uni.driver.js`
Old comopnents like `Calendar` export has 2 drivers. The old one is still exported, but as it didn't have `puppeteer` teskit, we are using the new `calendar.uni.driver` as the exported `puppeteer` testkit.

We decided to keep maintaining both kind of drivers for the old components, as it is too hard of a braking change for our consumers.

**What should we do next**
1. Old puppeteer and protractor testkits - i think we should keep them as they are, do not switch to new uni.driver, unless someone ask for some methods specifically.

2. Old components - like input - it is necessary to export besides to `InputTestkitFactory` also `InputTestkitFactoryAsync`? we thought it first but i don't see the benefit of that.
3. Suffix - is it necessary to distinguish between testkit and async testkit? use `async`? If we do it then what would happen to all the new components that their testkit is already of unidriver?
4. We have a practical usage docs 
 https://github.com/wix/wix-style-react/blob/b194a4f27bc9b48beced201c9d6d3383d315c92c/docs/usage/testing.md
https://github.com/wix/wix-style-react/blob/master/docs/usage/testing_examples.md

i think it's enough to just verify it's with async await in all example - without separation and confusing too much.

Basically it's kind of funny but in the bottom line i'm suggesting keeping everything as it is, let me know if you see it differently.

@shlomitc @argshook

**Comment by argshook** [comment]
# tl;dr;

`wix-style-react/dist/testkit/enzyme`:

```js
// current export
export const inputTestkitFactory = enzymeTestkitFactoryCreator(inputTestkitDriver)
export const tableTestkitFactory = enzymeUniTestkitFactoryCreator(inputTestkitDriver)
... hundreds like this
```

notice that `input` is not unidriver, but `table` is.

my suggestion

```diff
// current export
export const inputTestkitFactory = enzymeTestkitFactoryCreator(inputTestkitDriver)
export const tableTestkitFactory = enzymeUniTestkitFactoryCreator(inputTestkitDriver)
// ... hundreds like this

+export const InputTestkit = enzymeUniTestkitFactoryCreator(inputTestkitDriver)
+export const TableTestkit = enzymeUniTestkitFactoryCreator(inputTestkitDriver)
```

adding new entries, keeping old unchanged. All new entries are for unidriver, no guessing, methods always async.

---

# long read

my long dream of exposing testkits:

1. export testkits without the `Factory` word
2. capitalize first character to indicate it's a "factory"
3. all new exports are unidriver for all platforms

like this:

```diff
-inputTestkitFactory
+InputTestkit
```

all of the `ComponentNameFactory` exports would be only unidriver for
all platforms. No more confusion when, for example, enzyme driver is
exported as unidriver, but vanilla is legacy.

because name is different, we can still keep the old `componentNameTestkitFactory` without
a breaking change

we update docs to show only `InputTestkit` and new users get into the
habit. eventually we can choose to show deprecation log for
`inputTestkitFactory` users, but we don't have to.

this way we expose testkits that are always unidriver. we also don't
suddenly break existing users.

---

> 1. Old puppeteer and protractor testkits - i think we should keep them as they are, do not switch to new uni.driver, unless someone ask for some methods specifically.

with approach im describing we can add unidriver exports without a breaking change.

---

> 2. Old components - like input - it is necessary to export besides to InputTestkitFactory also InputTestkitFactoryAsync? we thought it first but i don't see the benefit of that.

as mentioned above, i would rather remove something from name than add.
We had discussions about `Async` suffix in the past, but i also don't
see how that would help

---

> 3. Suffix - is it necessary to distinguish between testkit and async testkit? use async?

not necesary, but i would argue that if only unidrivers would be exported with `ComponentNameTestkit` pattern, then there would be no confusion, as i mentioned with enzyme vs vanilla example above

> If we do it then what would happen to all the new components that their testkit is already of unidriver?

exports can be duplicated to avoid breaking change. Because testkit
exports are generated, we can easily have, say, `tableTestkitFactory`
and also `TableTestkit`.

---

sorry for wall of text, somehow i typed a lot

**Comment by shlomitc** [comment]
Regarding the specific old `puppeteer` and `protractor` testkits. another option will be to remove them completely by replacing them with `unidriver` and add to the unidriver methods which are specific to the old drivers (keep it for backward compatibility)

Regarding the `Async` annotation - it was meant to help to distinguish and emphasizing that all of the drivers should be consumed in an asynchronous approach (async-await, promises, etc...).  as long as we provide an isolated way to import each one - I'm fine with it

**Comment by mosheKerbel** [comment]
Cool, i agree with it all.
About this one: 

> Regarding the specific old puppeteer and protractor testkits. another option will be to remove them completely by replacing them with unidriver and add to the unidriver methods which are specific to the old drivers (keep it for backward compatibility)

I think it would be easier to just add the new ones instead of replacing, because you have components with many methods that exist in puppeteer and protractor old drivers, that don't exist in the new unidrivers, so it will be a tedious migration.

**Comment by shlomitc** [comment]
need to check if these drivers are even used - my guess is that they aren't

**Comment by EgoziE** [comment]
Most of it is done
